packages/console/src/containers/AppContent/components/Topbar/TenantSelector/index.tsx

```diff
@@ -0,0 +1,88 @@
+import classNames from 'classnames';
+import { useContext, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import KeyboardArrowDown from '@/assets/images/keyboard-arrow-down.svg';
+import PlusSign from '@/assets/images/plus.svg';
+import Tick from '@/assets/images/tick.svg';
+import AppError from '@/components/AppError';
+import Divider from '@/components/Divider';
+import Dropdown, { DropdownItem } from '@/components/Dropdown';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import { onKeyDownHandler } from '@/utils/a11y';
+
+import TenantEnvTag from './components/TenantEnvTag';
+import * as styles from './index.module.scss';
+
+function TenantSelector() {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { currentTenantId, tenants } = useContext(TenantsContext);
+  const [error, setError] = useState<Error>();
+  const anchorRef = useRef<HTMLDivElement>(null);
+  const [showDropdown, setShowDropdown] = useState(false);
+
+  const currentTenantInfo = useMemo(() => {
+    const tenant = tenants?.find((tenant) => tenant.id === currentTenantId);
+    if (!tenant) {
+      setError(new Error(`Can not find tenant with current tenant id: ${currentTenantId}`));
```

Comment:
i18n?

Reply:
p.s. Feels like we should have a high-level tenant guard to make sure current tenants data is not undefined

Reply:
Added an `useTenants` hook to have tenant-related data prepared.